matminer/featurizers/utils/roost.py

```diff
@@ -0,0 +1,308 @@
+      cif_id: str or int
+
+    Returns
+    -------
+    N = sum(n_i); N0 = sum(i)
+
+    batch_atom_weights: torch.Tensor shape (N, 1)
+    batch_atom_fea: torch.Tensor shape (N, orig_atom_fea_len)
+        Atom features from atom type
+    batch_self_fea_idx: torch.LongTensor shape (N, M)
+        Indices of mapping atom to copies of itself
+    batch_nbr_fea_idx: torch.LongTensor shape (N, M)
+        Indices of M neighbors of each atom
+    crystal_atom_idx: list of torch.LongTensor of length N0
+        Mapping from the crystal idx to atom idx
+    target: torch.Tensor shape (N, 1)
+        Target value for prediction
+    batch_comps: list
+    batch_ids: list
+    """
+    # define the lists
+    batch_atom_weights = []
+    batch_atom_fea = []
+    batch_self_fea_idx = []
+    batch_nbr_fea_idx = []
+    crystal_atom_idx = []
+    batch_target = []
+    batch_comp = []
+    batch_cry_ids = []
+
+    cry_base_idx = 0
+    for (
+        i,
+        ((atom_weights, atom_fea, self_fea_idx, nbr_fea_idx), target, comp, cry_id),
+    ) in enumerate(dataset_list):
+        # number of atoms for this crystal
+        n_i = atom_fea.shape[0]
+
+        # batch the features together
+        batch_atom_weights.append(atom_weights)
+        batch_atom_fea.append(atom_fea)
+
+        # mappings from bonds to atoms
+        batch_self_fea_idx.append(self_fea_idx + cry_base_idx)
+        batch_nbr_fea_idx.append(nbr_fea_idx + cry_base_idx)
+
+        # mapping from atoms to crystals
+        crystal_atom_idx.append(torch.tensor([i] * n_i))
+
+        # batch the targets and ids
+        batch_target.append(target)
+        batch_comp.append(comp)
+        batch_cry_ids.append(cry_id)
+
+        # increment the id counter
+        cry_base_idx += n_i
+
+    return (
+        (
+            torch.cat(batch_atom_weights, dim=0),
+            torch.cat(batch_atom_fea, dim=0),
+            torch.cat(batch_self_fea_idx, dim=0),
+            torch.cat(batch_nbr_fea_idx, dim=0),
+            torch.cat(crystal_atom_idx),
+        ),
+        torch.stack(batch_target, dim=0),
+        batch_comp,
+        batch_cry_ids,
+    )
+
+
+def init_roost(
+    model_name,
+    run_id,
+    task,
+    robust,
+    loss,
+    optim,
+    learning_rate,
+    weight_decay,
+    momentum,
+    device,
+    n_targets,
+    elem_emb_len,
+    elem_fea_len,
+    n_graph,
+    elem_heads,
+    elem_gate,
+    elem_msg,
+    cry_heads,
+    cry_gate,
+    cry_msg,
+    out_hidden,
+    fine_tune=None,
+):
+
+    if fine_tune is not None:
+        print(f"Use material_nn and output_nn from '{fine_tune}' as a starting point")
+        checkpoint = torch.load(fine_tune, map_location=device)
+        model = Roost(**checkpoint["model_params"], device=device,)
+        model.load_state_dict(checkpoint["state_dict"])
+
+        if model.model_params["robust"] == True:
+            raise NotImplementedError("Robust losses not supported within Matminer")
```

Comment:
left in to add `warm_restarts` although not yet configured. Will wait until a decision is made on where to store the default models.